File: app/src/main/java/io/github/shadow578/music_dl/downloader/DownloaderException.java
package io.github.shadow578.music_dl.downloader;

import androidx.annotation.NonNull;

import java.io.IOException;

import io.github.shadow578.music_dl.KtPorted;

/**
 * exception used by {@link DownloaderService}
 */
@KtPorted
public class DownloaderException extends IOException {

    public DownloaderException(@NonNull String message) {
        super(message);
    }

    public DownloaderException(@NonNull String message, @NonNull Throwable cause) {
        super(message, cause);
    }
}

File: app/src/main/kotlin/io/github/shadow578/yodel/downloader/DownloaderException.kt
package io.github.shadow578.yodel.downloader

import java.io.IOException

/**
 * exception used by [DownloaderService]
 */
class DownloaderException : IOException {
    /**
     * if this exception was thrown while downloading using youtube-dl, this value contains the
     * command line output of youtube-dl.
     * Otherwise, this is null
     */
    val downloaderOutput: String?

    constructor(message: String, dlOutput: String? = null) : super(message) {
        downloaderOutput = dlOutput
    }

    constructor(message: String, cause: Throwable, dlOutput: String? = null) : super(
        message,
        cause
    ) {
        downloaderOutput = dlOutput
    }
}